js/private/js_run_devserver.bzl

```diff
@@ -0,0 +1,95 @@
+"Implementation details for js_run_devserver rule"
+
+load(":js_binary.bzl", "js_binary_lib")
+load(":js_binary_helpers.bzl", _gather_files_from_js_providers = "gather_files_from_js_providers")
+load("@bazel_skylib//lib:dicts.bzl", "dicts")
+
+_DOC = """Runs a devserver via binary target or command.
+
+The devserver specified is run in a custom sandbox that is more compatible with
+node tools that come with watch mode such as Webpack devserver or Next.js.
+
+The custom sandbox is populated with the default outputs of all targets in `data`.
+rules_js node_modules links targets such as `//:node_modules/http-server` are supported.
+Since these targets are symlinks in the output tree, they are recreated as symlinks
+in the custom sandbox and do not incur a fully copy of the underlying npm packages.
+
+Supports running with ibazel. Only changes in `data` files are synchronized incrementally
+when running with ibazel.
+"""
+
+_attrs = dicts.add(js_binary_lib.attrs, {
+    "tool": attr.label(
+        doc = """The devserver binary target to run.
+        
+Only one of `command` or `tool` may be specified.""",
+        executable = True,
+        cfg = "exec",
+    ),
+    "command": attr.string(
+        doc = """The devserver command to run.
+
+For example, this could be an executable tool in an npm package that is included
```

Comment:
more guidance about how command fixes problems with several copies of react?

Reply:
sure. will add more to the docstring in the morning and add some examples as well